### PR TITLE
add `header_pages` config to link only specific files in header

### LINKED ISSUE #19: Ability to specify Pages in Header
It's awesome that the theme auto-includes a list of pages in the header, but once you get above a certain number of pages, it gets a bit unwieldy. 

It also be cool to be able to set the order of the pages. Thinking maybe some sort of array in the config of page paths to include, falling back to all pages if not set.

### PR DESCRIPTION
**Pull request by Crunch09**
fixes #19

### CODE REVIEW

**Comment by Crunch09** [comment]
cc: @benbalter

File: _includes/header.html [hunk 1]
```diff
@@ -4,6 +4,17 @@
 
     {% assign custom_url = site.url | append: site.baseurl %}
     {% assign full_base_url = custom_url | default: site.github.url %}
+    {% if site.header_pages %}
+      {% assign header_pages = "" | split:"|"  %}
```
Comment (benbalter) [inline _includes/header.html]: What's going on here?

File: _includes/header.html [hunk 2]
```diff
@@ -4,6 +4,17 @@
 
     {% assign custom_url = site.url | append: site.baseurl %}
     {% assign full_base_url = custom_url | default: site.github.url %}
+    {% if site.header_pages %}
```
Comment (benbalter) [inline _includes/header.html]: Could this be written as `{% assign header_pages = site.header_pages | default: site.pages %}`?

Comment (benbalter) [inline _includes/header.html]: I believe, if we do that, we'll only need to iterate through the pages once.

**Comment by benbalter** [comment]
Nice start! A few questions on the implementation to try to DRY things up a bit, but I really like it.

File: _includes/header.html (hunk 1 above)
Comment (Crunch09) [inline _includes/header.html]: I got this from https://talk.jekyllrb.com/t/how-do-you-add-items-to-an-array-in-jekyll/324/4 to initialize an empty array.

File: _includes/header.html (hunk 2 above)
Comment (Crunch09) [inline _includes/header.html]: Good suggestion! Yes we could do that,
but i think we should use something like this then:
```
{% assign default_paths = site.pages | map: "path" %}
{% assign header_pages = site.header_pages | default: default_paths %}
```
To make sure in both cases we're dealing with `Strings` and not in one case `Strings`, in the other one with `Page` objects.

In the remaining loop we could then convert these back  to `Pages` one by one. What do you think?

**Comment by Crunch09** [comment]
@benbalter Thanks for the review! Improved it now to use only one loop

**Comment by scoskey** [comment]
Up vote.

**Comment by benbalter** [comment]
Thanks @Crunch09! Sorry for the delay.

**Comment by Crunch09** [comment]
@benbalter No worries! Thank you for merging it.

**Pull request merged**